src/sidebar/components/hooks/test/use-user-filter-options-test.js

```diff
@@ -0,0 +1,149 @@
+    };
+
+    $imports.$mock({
+      '../../store/use-store': { useStoreProxy: () => fakeStore },
+    });
+
+    //mount(<DummyComponent />);
+  });
+
+  afterEach(() => {
+    $imports.$restore();
+  });
+
+  it('should return a user filter option for each user who has authored an annotation', () => {
+    fakeStore.allAnnotations.returns(annotationFixtures());
+
+    mount(<DummyComponent />);
+
+    const values = lastUserOptions.map(option => option.value);
+    const displays = lastUserOptions.map(option => option.display);
+
+    assert.deepEqual(values, [
+      'abalone',
+      'bananagram',
+      'acct:dingbat@localhost',
+    ]);
+    assert.deepEqual(displays, ['abalone', 'bananagram', 'dingbat']);
+  });
+
+  it('should use display names if feature flag enabled', () => {
+    fakeStore.allAnnotations.returns(annotationFixtures());
+    fakeStore.isFeatureEnabled.withArgs('client_display_names').returns(true);
+
+    mount(<DummyComponent />);
+
+    const values = lastUserOptions.map(option => option.value);
+    const displays = lastUserOptions.map(option => option.display);
+
+    assert.deepEqual(values, [
+      'abalone',
+      'acct:dingbat@localhost',
+      'bananagram',
+    ]);
+    assert.deepEqual(displays, ['Aba Lone', 'Ding Bat', 'Zerk']);
+  });
+
+  it('should add focused-user filter information if configured', () => {
+    fakeStore.allAnnotations.returns(annotationFixtures());
+    fakeStore.isFeatureEnabled.withArgs('client_display_names').returns(true);
+    fakeStore.getFocusFilters.returns({
+      user: { value: 'carrotNumberOne', display: 'Number One Carrot' },
+    });
+
+    mount(<DummyComponent />);
+
+    const values = lastUserOptions.map(option => option.value);
+    const displays = lastUserOptions.map(option => option.display);
+
+    assert.deepEqual(values, [
+      'abalone',
+      'acct:dingbat@localhost',
+      'carrotNumberOne',
+      'bananagram',
+    ]);
+    assert.deepEqual(displays, [
+      'Aba Lone',
+      'Ding Bat',
+      'Number One Carrot',
+      'Zerk',
+    ]);
+  });
+
+  it('always uses display name for focused user', () => {
+    fakeStore.allAnnotations.returns(annotationFixtures());
+    fakeStore.isFeatureEnabled.withArgs('client_display_names').returns(false);
+    fakeStore.getFocusFilters.returns({
+      user: { value: 'carrotNumberOne', display: 'Numero Uno Zanahoria' },
+    });
+
+    mount(<DummyComponent />);
+
+    const values = lastUserOptions.map(option => option.value);
+    const displays = lastUserOptions.map(option => option.display);
+
+    assert.deepEqual(values, [
+      'abalone',
+      'bananagram',
+      'acct:dingbat@localhost',
+      'carrotNumberOne',
+    ]);
+    assert.deepEqual(displays, [
+      'abalone',
+      'bananagram',
+      'dingbat',
+      'Numero Uno Zanahoria',
+    ]);
+  });
+
+  it('overrides user filter option with user-focus filter option if there is a conflict', () => {
```

Comment:
Where is the "user filter option" referred to here being set?

Reply:
Bah, it's really hard to phrase this concisely (see my very wordy explanation of what's going on here). What I'm trying to summarize is: "In the case where focused user is defined and a filter option for that user already exists because they've also authored an annotation in the current annotation set, prefer the focus-filter variant over the annotation-user one." Swear word. Help.

Reply:
For now: `it('resolves user duplicates in favor of the focused user')`